### PR TITLE
add error message when cassete file does not exist

### PR DESCRIPTION
**Pull request by PauloVilarinho**
Issue: https://github.com/vcr/vcr/issues/985

Adds error message to when cassete is using record mode :none but file is not found

### CODE REVIEW

File: lib/vcr/errors.rb [hunk 1]
```diff
@@ -212,6 +212,14 @@ def format_foot_note(url, index)
           "https://benoittgt.github.io/vcr/?v=%s#/record_modes/none"
         ],
 
+        :none_without_file => [
+          ["The current record mode (:none) does not allow requests to be recorded.",
+           "One or more cassete names registered was not found. Use ",
```
Comment (olleolleolle) [inline lib/vcr/errors.rb:217]:
```suggestion
           "One or more cassette names registered was not found. Use ",
```

File: spec/lib/vcr/errors_spec.rb [hunk 2]
```diff
@@ -120,6 +120,12 @@ def request_with(options)
           end
         end
 
+        it 'mentions that one of the files does not exist when using :none and file does not exist' do
+          VCR.use_cassette('example', :record => :none) do
+            expect(message).to include('One or more cassete names registered was not found.')
```
Comment (olleolleolle) [inline spec/lib/vcr/errors_spec.rb:125]:
```suggestion
            expect(message).to include('One or more cassette names registered was not found.')
```

File: lib/vcr/errors.rb [hunk 3]
```diff
@@ -284,14 +292,22 @@ def record_mode_suggestion
...
+          none_suggestion
         elsif record_modes.all?{|r| r == :once }
           [:delete_cassette_for_once]
         else
           []
         end
       end
 
+      def none_suggestion
+        if current_cassettes.any?{|c| !File.exist?(c.file) }
```
Comment (olleolleolle) [inline lib/vcr/errors.rb:304]:
```suggestion
        if current_cassettes.any? {|c| !File.exist?(c.file) }
```

**Review by olleolleolle — COMMENTED** [review]
Thank you, @PauloVilarinho, I added some suggestions, this looks like we can get it in next week!

**Comment by PauloVilarinho** [comment]
All changes applied @olleolleolle, thanks for the feedback!

File: lib/vcr/errors.rb [hunk 4]
```diff
@@ -284,14 +292,22 @@ def record_mode_suggestion
...
         else
           []
         end
       end
 
+      def none_suggestion
+        if current_cassettes.any? {|c| !File.exist?(c.file) }
+          [:none_without_file]
+        else
+          [:deal_with_none]
```
Comment (andrehjr) [inline lib/vcr/errors.rb:307]: Should we add something for this scenario when a file does exist?

Running the suite locally and checking the coverage, this line is unreachable now.

image

Comment (PauloVilarinho) [reply lib/vcr/errors.rb:307]: Added tests for this, nice catch!

**Review by olleolleolle — APPROVED** [review]
Thank you, @PauloVilarinho!

**Pull request merged**